### ISSUE TITLE
Node synchronization protocol is broken

### ISSUE THREAD

**Issue by OlegMazurov** [hashgraph/hedera-services#14864]
### Description

A read operation during node synchronization times out and the connection gets broken. Stack traces from exceptions on the two sides suggest that both nodes were waiting for each other in different .
`node23` sees a timeout while reading _tips_:
```
2024-08-14 21:59:42.455 6652     WARN  SOCKET_EXCEPTIONS <<platform-core: SyncProtocolWith10 23 to 10>> NetworkUtils: Connection broken: 23 <- 10
java.io.IOException: com.swirlds.common.threading.pool.ParallelExecutionException: Time thrown: 2024-08-14T21:59:42.454648407Z
        at com.swirlds.platform.gossip.sync.protocol.SyncProtocol.runProtocol(SyncProtocol.java:273) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.network.communication.states.ProtocolNegotiated.transition(ProtocolNegotiated.java:62) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.network.communication.Negotiator.execute(Negotiator.java:94) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.network.communication.ProtocolNegotiatorThread.run(ProtocolNegotiatorThread.java:85) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.doWork(StoppableThreadImpl.java:614) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.run(StoppableThreadImpl.java:215) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: com.swirlds.common.threading.pool.ParallelExecutionException: Time thrown: 2024-08-14T21:59:42.454648407Z
        at com.swirlds.common.threading.pool.CachedPoolParallelExecutor.doParallel(CachedPoolParallelExecutor.java:101) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.readWriteParallel(ShadowgraphSynchronizer.java:468) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.reserveSynchronize(ShadowgraphSynchronizer.java:254) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.synchronize(ShadowgraphSynchronizer.java:193) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.sync.protocol.SyncProtocol.runProtocol(SyncProtocol.java:270) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        ... 6 more
Caused by: java.net.SocketTimeoutException: Read timed out
        at java.base/sun.nio.ch.NioSocketImpl.timedRead(NioSocketImpl.java:278) ~[?:?]
        at java.base/sun.nio.ch.NioSocketImpl.implRead(NioSocketImpl.java:304) ~[?:?]
        at java.base/sun.nio.ch.NioSocketImpl.read(NioSocketImpl.java:346) ~[?:?]
        at java.base/sun.nio.ch.NioSocketImpl$1.read(NioSocketImpl.java:796) ~[?:?]
        at java.base/java.net.Socket$SocketInputStream.read(Socket.java:1099) ~[?:?]
        at java.base/sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:489) ~[?:?]
        at java.base/sun.security.ssl.SSLSocketInputRecord.readHeader(SSLSocketInputRecord.java:483) ~[?:?]
        at java.base/sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:70) ~[?:?]
        at java.base/sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1461) ~[?:?]
        at java.base/sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:1066) ~[?:?]
        at com.swirlds.common.io.extendable.extensions.AbstractStreamExtension.read(AbstractStreamExtension.java:88) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.common.io.extendable.ExtendableInputStream.read(ExtendableInputStream.java:78) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at java.base/java.io.BufferedInputStream.fill(BufferedInputStream.java:291) ~[?:?]
        at java.base/java.io.BufferedInputStream.read1(BufferedInputStream.java:347) ~[?:?]
        at java.base/java.io.BufferedInputStream.implRead(BufferedInputStream.java:420) ~[?:?]
        at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:399) ~[?:?]
        at java.base/java.io.DataInputStream.readFully(DataInputStream.java:208) ~[?:?]
        at java.base/java.io.DataInputStream.readInt(DataInputStream.java:385) ~[?:?]
        at com.swirlds.common.io.streams.AugmentedDataInputStream.readInt(AugmentedDataInputStream.java:169) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.common.io.streams.AugmentedDataInputStream.readBooleanList(AugmentedDataInputStream.java:346) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.SyncUtils.lambda$readMyTipsTheyHave$7(SyncUtils.java:170) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.common.threading.pool.CachedPoolParallelExecutor.doParallel(CachedPoolParallelExecutor.java:99) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.readWriteParallel(ShadowgraphSynchronizer.java:468) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.reserveSynchronize(ShadowgraphSynchronizer.java:254) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.synchronize(ShadowgraphSynchronizer.java:193) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.sync.protocol.SyncProtocol.runProtocol(SyncProtocol.java:270) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        ... 6 more
```
`node10` advanced to next step in the protocol and was blocked on flushing events:
```
2024-08-14 21:59:42.472 6654     WARN  SOCKET_EXCEPTIONS <<platform-core: SyncProtocolWith23 10 to 23>> NetworkUtils: Connection broken: 10 -> 23
java.io.IOException: com.swirlds.common.threading.pool.ParallelExecutionException: Time thrown: 2024-08-14T21:59:42.472040153Z
        at com.swirlds.platform.gossip.sync.protocol.SyncProtocol.runProtocol(SyncProtocol.java:273) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.network.communication.states.ProtocolNegotiated.transition(ProtocolNegotiated.java:62) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.network.communication.Negotiator.execute(Negotiator.java:94) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.network.communication.ProtocolNegotiatorThread.run(ProtocolNegotiatorThread.java:85) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.doWork(StoppableThreadImpl.java:614) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.run(StoppableThreadImpl.java:215) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: com.swirlds.common.threading.pool.ParallelExecutionException: Time thrown: 2024-08-14T21:59:42.472040153Z
        at com.swirlds.common.threading.pool.CachedPoolParallelExecutor.doParallel(CachedPoolParallelExecutor.java:101) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.readWriteParallel(ShadowgraphSynchronizer.java:468) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.sendAndReceiveEvents(ShadowgraphSynchronizer.java:406) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.reserveSynchronize(ShadowgraphSynchronizer.java:271) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.synchronize(ShadowgraphSynchronizer.java:193) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.sync.protocol.SyncProtocol.runProtocol(SyncProtocol.java:270) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        ... 6 more
        Suppressed: java.util.concurrent.ExecutionException: java.net.SocketException: Connection or outbound has closed
                at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[?:?]
                at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[?:?]
                at com.swirlds.common.threading.pool.CachedPoolParallelExecutor.doParallel(CachedPoolParallelExecutor.java:106) ~[swirlds-common-0.53.0-alpha.4.jar:?]
                at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.readWriteParallel(ShadowgraphSynchronizer.java:468) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.sendAndReceiveEvents(ShadowgraphSynchronizer.java:406) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.reserveSynchronize(ShadowgraphSynchronizer.java:271) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.synchronize(ShadowgraphSynchronizer.java:193) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at com.swirlds.platform.gossip.sync.protocol.SyncProtocol.runProtocol(SyncProtocol.java:270) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at com.swirlds.platform.network.communication.states.ProtocolNegotiated.transition(ProtocolNegotiated.java:62) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at com.swirlds.platform.network.communication.Negotiator.execute(Negotiator.java:94) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at com.swirlds.platform.network.communication.ProtocolNegotiatorThread.run(ProtocolNegotiatorThread.java:85) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.doWork(StoppableThreadImpl.java:614) ~[swirlds-common-0.53.0-alpha.4.jar:?]
                at com.swirlds.common.threading.framework.internal.StoppableThreadImpl.run(StoppableThreadImpl.java:215) ~[swirlds-common-0.53.0-alpha.4.jar:?]
                at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
        Caused by: java.net.SocketException: Connection or outbound has closed
                at java.base/sun.security.ssl.SSLSocketImpl$AppOutputStream.write(SSLSocketImpl.java:1297) ~[?:?]
                at com.swirlds.common.io.extendable.extensions.AbstractStreamExtension.write(AbstractStreamExtension.java:130) ~[swirlds-common-0.53.0-alpha.4.jar:?]
                at com.swirlds.common.io.extendable.ExtendableOutputStream.write(ExtendableOutputStream.java:79) ~[swirlds-common-0.53.0-alpha.4.jar:?]
                at java.base/java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:125) ~[?:?]
                at java.base/java.io.BufferedOutputStream.implFlush(BufferedOutputStream.java:252) ~[?:?]
                at java.base/java.io.BufferedOutputStream.flush(BufferedOutputStream.java:240) ~[?:?]
                at java.base/java.io.DataOutputStream.flush(DataOutputStream.java:131) ~[?:?]
                at com.swirlds.platform.gossip.shadowgraph.SyncUtils.lambda$sendEventsTheyNeed$8(SyncUtils.java:233) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
                at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
                at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
                at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
                ... 1 more
Caused by: java.io.EOFException
        at java.base/java.io.DataInputStream.readUnsignedByte(DataInputStream.java:297) ~[?:?]
        at java.base/java.io.DataInputStream.readByte(DataInputStream.java:275) ~[?:?]
        at com.swirlds.common.io.streams.AugmentedDataInputStream.readByte(AugmentedDataInputStream.java:144) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.SyncUtils.lambda$readEventsINeed$9(SyncUtils.java:273) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.common.threading.pool.CachedPoolParallelExecutor.doParallel(CachedPoolParallelExecutor.java:99) ~[swirlds-common-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.readWriteParallel(ShadowgraphSynchronizer.java:468) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.sendAndReceiveEvents(ShadowgraphSynchronizer.java:406) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.reserveSynchronize(ShadowgraphSynchronizer.java:271) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.shadowgraph.ShadowgraphSynchronizer.synchronize(ShadowgraphSynchronizer.java:193) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        at com.swirlds.platform.gossip.sync.protocol.SyncProtocol.runProtocol(SyncProtocol.java:270) ~[swirlds-platform-core-0.53.0-alpha.4.jar:?]
        ... 6 more
``` 

### Steps to reproduce

The issue is seen a lot during longevity testing in `perf1`.

### Hedera network

other

### Version

0.53.0-alpha.4

### Operating system

Linux

**Comment by lpetrovic05** [comment]
Since node `23` timed out while reading booleans, and node `10` was already doing the next step, my only explanation would be that there was a network disruption. Node `10` wrote the booleans to the socket and continued to read an write events. Node `23` never received these booleans, so it just timed out waiting.

**Comment by cody-littley** [comment]
I agree with Lazar. This fingerprint matches a network disruption.

**Comment by OlegMazurov** [comment]
No, I have not had time to investigate it further but I would agree it's a low priority issue until proven otherwise.

**Comment by poulok** [comment]
@OlegMazurov may I close the ticket now or do you still plan on looking into this?

**Comment by abies** [comment]
This was happening in the code which is not longer present (got replaced with RPC Sync).
